doc/contribute/doc-guidelines.rst

```diff
@@ -0,0 +1,308 @@
+.. _doc_guidelines:
+
+Documentation Guidelines for the Zephyr Project
+###############################################
+
+Zephyr Project content is written using the `reStructuredText`_ markup
+language (.rst file extension) with Sphinx extensions, and processed
+using sphinx to create a formatted standalone website.  Developers can
+view this content either in its raw form as .rst markup files, or (with
+sphinx installed) they can run the makefile (or make.bat on windows) to
+generate the HTML content and view it with a web browser directly from
+their workstation's drive.  This same .rst content is also fed into the
+`Zephyr documentation`_ website (with a different theme applied).
+
+You can read details about `reStructuredText`_
+and about `Sphinx extensions`_ from their respective websites.
+
+.. _Sphinx extensions: http://www.sphinx-doc.org/en/stable/contents.html
+.. _reStructuredText: http://docutils.sourceforge.net/docs/ref/rst/restructuredtext.html
+.. _Sphinx Inline Markup:  http://sphinx-doc.org/markup/inline.html#inline-markup
+.. _Zephyr documentation:  http://docs.zephyrproject.org
+
+This document provides a quick reference for commonly used ReST and
+sphinx-defined directives and roles used to create the documentation
+you're reading.
+
+Headings
+********
+
+For consistency, please use just an underline indicator for headings.
+
+* Document title (h1) use "#" for the underline character
+* First section heading level (h2) use "*"
+* Second section heading level (h3) use "="
+* Third section heading level (h4) use "-"
+
+The heading underline must be at least as long as the title it's under.
+
+Content Highlighting
+********************
+
+Some common ReST inline markup samples:
+
+* one asterisk: ``*text*`` for emphasis (*italics*),
+* two asterisks: ``**text**`` for strong emphasis (**boldface**), and
+* backquotes: ````text```` for inline code samples.
+
+If asterisks or backquotes appear in running text and could be confused with
+inline markup delimiters, you can eliminate the confusion by adding a
+backslash ("\\") before it.
+
+Lists
+*****
+
+For bullet lists, place an asterisk (``*``) or hyphen (``-``) at
+the start of a paragraph and indent continuation lines with three
```

Comment:
Should be two spaces, right?

The usual rule I see applied is to align continuation lines with the first non-whitespace character following the initial hyphen; that's two spaces, no? Following this rule as written would result in odd indentation in the source file:

```
- Here is the beginning of a multi-line bullet item, showing why
   three spaces of indent results in an odd continuation line;
   none of the continued lines align with the initial line
```